appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartDataModel.java

```diff
@@ -0,0 +1,222 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import org.pepstock.charba.client.data.Data;
+import org.pepstock.charba.client.data.Dataset;
+
+import java.util.List;
+
+/**
+ * Chart Data Model base class.
+ *
+ * The Data Model is responsible for handling data operations
+ * of a Chart, and represents the data of a single Data Series.
+ * Individual styling and options of a single Data Series is also
+ * a responsibility of the Data Model class.
+ */
+public abstract class MockChartDataModel<D extends Dataset> {
+    protected D dataSeries;
+    protected Data chartData;
+
+    /**
+     * Creates a new Mock Chart Model object instance, linking it with
+     * the Data object of a specific Chart.
+     *
+     * @param chartData  Chart Data object to link to
+     */
+    protected MockChartDataModel(Data chartData) {
+        this.chartData = chartData;
+    }
+
+    /**
+     * Changes the Color of the Data Series.
+     *
+     * @param color  New Color value in &HAARRGGBB format.
+     */
+    public abstract void changeColor(String color);
+
+    /**
+     * Changes the label of the Data Series.
+     *
+     * @param text  New text value
+     */
+    public void changeLabel(String text) {
+        dataSeries.setLabel(text);
+    }
+
+    /**
+     * Adds the data series of this object to the Chart.
+     */
+    protected void addDataSeriesToChart() {
+        // When adding the first Data Series, it should be set
+        // to the Chart Data object itself rather then appended,
+        // to register the first (new) DataSet List to the Chart data.
+        // Subsequent adding of Data Series objects can simply be added
+        // to the end of the List.
+        if (chartData.getDatasets().size() == 0) {
+            chartData.setDatasets(dataSeries);
+        } else {
+            chartData.getDatasets().add(dataSeries);
+        }
+    }
+
+    /**
+     * Removes the Data Series from the Chart.
+     */
+    public void removeDataSeriesFromChart() {
+        chartData.getDatasets().remove(dataSeries);
+    }
+
+    /**
+     * Converts an ARGB color to RGB hex format.
+     *
+     * @param color  &HAARRGGBB format color string
+     * @return #RRGGBB format color string
+     */
+    protected String getHexColor(String color) {
+        // The idea: Remove &H at the beginning, replace with # and reorder ARGB to RGB
+        return "#" + color.substring(4);
+    }
+
+    /**
+     * Sets the elements of the Data Series from a CSV-formatted String.
+     *
+     * @param elements String in CSV format
+     */
+    public void setElements(String elements) {
+        // Split the entries by the comma
+        // TODO: Possibly move this to the server side? (commas not escaped with this approach)
```

Comment:
@ewpatton If I recall correctly, there were a few issues here. The first is namely that when using the CsvParser in appengine/server, exceptions would be thrown on the mock side because of certain packages not being defined for the client side of appengine (I think it might have been the regex package..) [I would have to double check on this one]

It might make sense to move some functionality away from the client CSVUtil, though, although it's primarily focused for YailLists.